From 558f56903748408bc07aa1a0b9c8b488f97d7fb4 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Fri, 7 Aug 2026 12:58:12 +0200
Subject: [PATCH 1/2] FIX: Fix libslic3r_cgal not depending on Cereal

libslic3r_cgal uses Cereal's header-only library but didn't depend on
it. This means that one of the other dependencies (libigl, mcut or cgal)
must be installed in the same location for the compiler to be able to
find the header.

Add the missing dependency to avoid this problem.
---
 src/libslic3r/CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/libslic3r/CMakeLists.txt b/src/libslic3r/CMakeLists.txt
index 46cd7970a373..2667c494910d 100644
--- a/src/libslic3r/CMakeLists.txt
+++ b/src/libslic3r/CMakeLists.txt
@@ -577,7 +577,8 @@ if (CMAKE_COMPILER_IS_GNUCC AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 14.1)
     set_property(TARGET libslic3r_cgal PROPERTY CXX_EXTENSIONS OFF)
 endif()
 
-target_link_libraries(libslic3r_cgal PRIVATE ${_cgal_tgt} libigl mcut)
+find_package(cereal REQUIRED)
+target_link_libraries(libslic3r_cgal PRIVATE ${_cgal_tgt} libigl mcut cereal)
 if(SLIC3R_HAS_OPENMESHCRAFT_BOOLEAN)
     # GMP/MPFR usually come with CGAL/DEP, but may not be visible to find_library
     # in non-standard layouts. Missing deps used to fall back to MCUT silently;
-- 
2.55.0


From 4b5cf4181307a44e2c2777bfad43c777de6adf96 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Wed, 5 Aug 2026 00:19:18 +0200
Subject: [PATCH 2/2] ENH: Allow using system CGAL

To make it easier to test newer versions, or apply system specific
patches.
---
 deps/CMakeLists.txt | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/deps/CMakeLists.txt b/deps/CMakeLists.txt
index 2fb2282a85a9..f426a0b2ce0a 100644
--- a/deps/CMakeLists.txt
+++ b/deps/CMakeLists.txt
@@ -45,6 +45,7 @@ option(DEP_BUILD_FREETYPE "Compile freetype" ON)
 option(DEP_BUILD_WXWIDGETS "Compile wxWidgets" ON)
 option(DEP_BUILD_GMP "Compile GMP" ON)
 option(DEP_BUILD_MPFR "Compile MPFR" ON)
+option(DEP_BUILD_CGAL "Compile CGAL" ON)
 option(DEP_BUILD_FFMPEG "Compile ffmpeg" ON)
 option(DEP_BUILD_LIBHARU "Compile libharu" ON)
 
@@ -268,7 +269,11 @@ if (DEP_BUILD_MPFR)
     include(MPFR/MPFR.cmake)
     set(MPFR_PKG dep_MPFR)
 endif ()
-include(CGAL/CGAL.cmake)
+set(CGAL_PKG "")
+if (DEP_BUILD_CGAL)
+    include(CGAL/CGAL.cmake)
+    set(CGAL_PKG dep_CGAL)
+endif()
 
 include(NLopt/NLopt.cmake)
 
@@ -317,7 +322,7 @@ set(_dep_list
     dep_OpenVDB
     dep_OpenCSG
     dep_OpenCV
-    dep_CGAL
+    ${CGAL_PKG}
     ${OPENSSL_PKG}
     ${GLFW_PKG}
     ${PNG_PKG}
-- 
2.55.0

